### PR TITLE
HZN-1468: Remove features-maven-plugin from the build

### PR DESCRIPTION
**Pull request by mattixtech**
This PR removes the usage of the features-maven-plugin (for java 9 compatibility) in favour of a handcrafted features.xml.

Besides the automated tests, I've verified product install and the system directory / features.xml look sane. Let me know if there is anything else we should verify before this PR can be merged.

* JIRA: http://issues.opennms.org/browse/HZN-1468

Our [continuous integration system](http://bamboo.internal.opennms.com:8085) will test and verify your changes.

### CODE REVIEW

File: container/features/src/main/resources/features.xml [hunk 1]
```diff
@@ -1,1333 +1,1155 @@
...
-    </feature>
-
-    <feature name="commons-net" description="Apache :: commons-net" version="3.6">
-      <bundle>mvn:commons-net/commons-net/3.6</bundle>
-    </feature>
-
-    <feature name="dnsjava" description="dnsjava" version="${dnsjavaVersion}">
-      <bundle dependency="true">wrap:mvn:org.dnsjava/dnsjava/${dnsjavaVersion}</bundle>
+    <repository>mvn:org.apache.activemq/activemq-karaf/${activemqVersion}/xml/features</repository>
+    <!--TODO: This could be moved to the full assembly pom-->
```
Comment (j-white) [inline container/features/src/main/resources/features.xml]: TODO

File: container/features/src/main/resources/features.xml [hunk 2]
```diff
@@ -1,1333 +1,1155 @@
...
+        <feature>alarm-change-notifier</feature>
+        <feature>opennms-es-rest</feature>
+        <feature>internal-plugins-descriptor</feature>
+        <feature>ifttt-integration</feature>
+        <feature>opennms-flows</feature>
+        <feature>opennms-api-layer</feature>
+        <feature>karaf-extender</feature>
+        <feature>opennms-alarm-history-elastic</feature>
+    </feature>
+    <feature name="opennms-integration-api" version="${opennmsApiVersion}" description="OpenNMS Integration API">
```
Comment (j-white) [inline container/features/src/main/resources/features.xml]: Can we remove this feature, and the sample project one, and reference them from the `<repository>mvn:org.opennms.integration.api/karaf-features/${opennmsApiVersion}/xml</repository>` instead?

File: container/features/src/main/resources/features.xml (hunk 1 above)
Comment (j-white) [reply container/features/src/main/resources/features.xml]: I'm in favor of removing this mega feature and declaring the features to be pulled into the system directory in OpenNMS directly in the full assembly pom instead.

**Comment by RangerRick** [comment]
I've added this branch to dependency monkey: https://bamboo.opennms.org/browse/OPENNMS-DM15-1

**Comment by mattixtech** [comment]
My latest changes fix the error seen in the dependency monkey run. I've also attempted to find any similar problems and fix them. We'll have to see what it looks like next run.

File: container/features/src/main/resources/features.xml [hunk 3]
```diff
@@ -1352,17 +982,499 @@
...
+    </feature>
+    <feature name="jax-rs-provider-multipart" version="${osgiJaxRsVersion}">
+        <bundle>mvn:com.eclipsesource.jaxrs/provider-multipart/${osgiJaxRsVersion}</bundle>
+    </feature>
+    <feature name="jax-rs-provider-security" version="${osgiJaxRsVersion}">
+        <bundle>mvn:com.eclipsesource.jaxrs/provider-security/${osgiJaxRsVersion}</bundle>
+    </feature>
+    <feature name="jax-rs-provider-sse" version="${osgiJaxRsVersion}">
+        <bundle>mvn:com.eclipsesource.jaxrs/provider-sse/${osgiJaxRsVersion}</bundle>
+    </feature>
```
[the remark this answers is not in the document]
Comment (mattixtech) [reply container/features/src/main/resources/features.xml]: Yes, good catch.

**Comment by mattixtech** [comment]
Latest changes aren't building on bamboo, looking into it...

**Comment by mattixtech** [comment]
Latest change should resolve the build issue.

**Comment by j-white** [comment]
Comments have been addressed, and dependency monkey is happy. Anything else here before we merge?

**Comment by mattixtech** [comment]
> Comments have been addressed, and dependency monkey is happy. Anything else here before we merge?

Good to merge.

**Pull request merged**